numpy/core/arrayprint.py

```diff
@@ -427,12 +426,15 @@ def wrapper(self, *args, **kwargs):
 # gracefully handle recursive calls, when object arrays contain themselves
 @_recursive_guard()
 def _array2string(a, options, separator=' ', prefix=""):
+    # the formatters can't deal with the way that np.matrix refuses to be
+    # made 1-d
+    data = asarray(a)
```

Comment:
Arguably as a result of this change, `_leading_trailing` didn't need fixing - but the cleanup above seems better anyway.

Reply:
I'm actually surprised this wasn't up here before. I would have thought `_get_format_function` would complain about getting subclasses, and only when the array size was large.

Reply:
In order for this to matter, you had to pass an array where `size > threshold`, but no single axis is larger than `2 * edgeitems`. I doubt that was tested.